File: app/src/main/java/com/uvnode/typhoon/extensions/api/ApiResponse.java
package com.uvnode.typhoon.extensions.api;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class ApiResponse {

    @Retention(RetentionPolicy.SOURCE)
    @IntDef({NONE, SOME})
    public @interface Result {}

    public static final int NONE = 0;
    public static final int SOME = 1;

    private Object payload;
    private int responseResult;

    public ApiResponse(int result) {
        this.responseResult = result;
    }

    public int getResult() {
        return responseResult;
    }

    public Object get() {
        return payload;
    }

    public void set(Object payload) {
        this.payload = payload;
    }
}

File: app/src/main/java/com/uvnode/typhoon/extensions/api/ApiResponse.kt
package com.uvnode.typhoon.extensions.api

class ApiResponse(val result: Int) {
    private var payload: Any? = null
    fun get(): Any? {
        return payload
    }

    fun set(payload: Any?) {
        this.payload = payload
    }

    companion object {
        const val NONE = 0
        const val SOME = 1
    }
}